src/rtctools/optimization/homotopy_mixin.py

```diff
@@ -137,24 +137,35 @@ def optimize(self, preprocessing=True, postprocessing=True, log_solver_failure_a
 
             else:
                 if self.__theta == options["theta_start"]:
-                    break
-
-                self.__theta -= delta_theta
-                delta_theta /= 2
-
-                if delta_theta < options["delta_theta_min"]:
-                    failure_message = (
-                        "Solver failed with homotopy parameter theta = {}. Theta cannot "
-                        "be decreased further, as that would violate the minimum delta "
-                        "theta of {}.".format(self.__theta, options["delta_theta_min"])
-                    )
-                    if log_solver_failure_as_error:
-                        logger.error(failure_message)
+                    # if an imported seed was used and theta != 0.0 then we fallback to using
+                    # default homotopy approach
+                    if self.seeding_options()["import_seed"] and 0.0 < options["theta_start"]:
+                        self.seeding_failed = True
+                        self.__theta = -delta_theta
```

Comment:
Do you want to decrease self.__theta if it already equals theta_start? I would expect you just keep it unchanged in this particular case.